### PR TITLE
Integrate with mh-start-with-module

### PR DESCRIPTION
**Pull request by dvdplm**
Integrate with mh-start-with-module that changes the way modules are added to the server.

### CODE REVIEW

File: client/rpc-servers/src/lib.rs [hunk 1]
```diff
@@ -91,24 +91,23 @@ mod inner {
...
 				};
-
+				// TODO: (dp) DRY this up; it's the same as the WS code
 				let handle = server.stop_handle();
-
-				server.register_module(module).expect("infallible already checked; qed");
 				let mut methods_api = RpcModule::new(());
-				let mut methods = server.method_names();
-				methods.sort();
+				let mut available_methods = module.method_names().collect::<Vec<&'static str>>();
```
Comment (maciejhirsz) [inline client/rpc-servers/src/lib.rs:97]: `collect` only needs to know the container type, item will get inferred.

```suggestion
				let mut available_methods = module.method_names().collect::<Vec<_>>();
```

File: client/rpc-servers/src/lib.rs [hunk 2]
```diff
@@ -158,23 +157,23 @@ mod inner {
...
 					}
 				};
-
+				// TODO: (dp) DRY this up; it's the same as the HTTP code
 				let handle = server.stop_handle();
-				server.register_module(module).expect("infallible already checked; qed");
 				let mut methods_api = RpcModule::new(());
-				let mut methods = server.method_names();
-				methods.sort();
+				let mut available_methods = module.method_names().collect::<Vec<&str>>();
```
Comment (maciejhirsz) [inline client/rpc-servers/src/lib.rs:163]: Ditto.
```suggestion
				let mut available_methods = module.method_names().collect::<Vec<_>>();
```

**Review by maciejhirsz — APPROVED** [review]
Would comment on not duplicating methods logic, but you already have that in a TODO :)

File: client/rpc-servers/src/lib.rs [hunk 3]
```diff
@@ -158,23 +157,23 @@ mod inner {
...
+				// TODO: (dp) DRY this up; it's the same as the HTTP code
 				let handle = server.stop_handle();
-				server.register_module(module).expect("infallible already checked; qed");
 				let mut methods_api = RpcModule::new(());
-				let mut methods = server.method_names();
-				methods.sort();
+				let mut available_methods = module.method_names().collect::<Vec<_>>();
+				available_methods.sort();
 
+				// TODO: (dp) not sure this is correct; shouldn't the `rpc_methods` also be listed?
```
Comment (maciejhirsz) [inline client/rpc-servers/src/lib.rs:166]: It's not listed on current live.

Comment (dvdplm) [reply client/rpc-servers/src/lib.rs:166]: Oh it isn't? Weird.

File: client/rpc-servers/src/lib.rs [hunk 4]
```diff
@@ -91,24 +91,23 @@ mod inner {
...
-
+				// TODO: (dp) DRY this up; it's the same as the WS code
 				let handle = server.stop_handle();
-
-				server.register_module(module).expect("infallible already checked; qed");
 				let mut methods_api = RpcModule::new(());
-				let mut methods = server.method_names();
-				methods.sort();
+				let mut available_methods = module.method_names().collect::<Vec<_>>();
+				available_methods.sort();
```
Comment (maciejhirsz) [inline client/rpc-servers/src/lib.rs:98]: Something that just popped in my head: there are no duplicate names here, and even if there were we wouldn't care about preserving their ordering, so unstable sort is faster.

```suggestion
				available_methods.sort_unstable();
```

**Pull request merged**